### PR TITLE
Add tests and more

### LINKED ISSUE #44: Remove mandatory environment file
Due to a limitation of newman the only way to pass variables in during run time in TFS is using global variables. 
When we trigger a collection and pass all variables in using tfs, no environment files are used.
It would be nice if the environment file was not mandatory as in this scenario its not required.

### LINKED ISSUE #46: Specify collection using URL
I was hoping to be able to specify my collection using a postman provided URL such as: https://www.getpostman.com/collections/secret_hex_guid_here . I tried placing the URL in the "Collection File Source" field, but no luck. Error was: "##[error]Not found collectionFileSource: D:\a\r1\a\https:\www.getpostman.com\collections\cd..." Is what I am trying to do supported?

### PR DESCRIPTION
**Pull request by sebcaps**
This PR aims to : 
- add tests to the task
- fix #44 
- fix #46 

Please do not consider PR #45, since this one handles it.
I assume all unit test are Ok, but no clues if it will work with newman (especially URL's support) but according to https://github.com/carlowahlstedt/NewmanPostman_VSTS_Task/issues/46#issuecomment-444341904, it should work like a charm...
I added radio to specify if one use file url or file, default is file for compatibility

### CODE REVIEW

**Comment by carlowahlstedt** [comment]
This is some great work!

I've added a few comments for things to consider. Additionally, the build is failing because:

> 2018-12-09T20:27:55.4665359Z > vstsexttask@0.0.1 compile D:\a\1\s
2018-12-09T20:27:55.4665699Z > tsc -p .
2018-12-09T20:27:55.4665799Z 
2018-12-09T20:27:55.4665923Z Tests/customReporterTest.ts(29,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4666072Z Tests/emptyCollectionTest.ts(25,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4666205Z Tests/emptyContents.ts(25,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4666367Z Tests/emptyEnvironmentFile.ts(24,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4666515Z Tests/emptyEnvironmentURL.ts(24,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4666638Z Tests/folderTest.ts(29,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4666793Z Tests/globalVarFileTest.ts(30,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4666919Z Tests/globalVarTest.ts(29,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4667041Z Tests/noEnvironment.ts(24,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4667180Z Tests/numberOfIterationTest.ts(29,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4667304Z Tests/useCollectionURL.ts(25,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4667449Z Tests/useEnvironmentFile.ts(30,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4667573Z Tests/useEnvironmentURL.ts(26,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4668726Z Tests/wrongCollectionURL.ts(25,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.
2018-12-09T20:27:55.4668884Z Tests/wrongEnvironmentURL.ts(25,9): error TS2339: Property 'stats' does not exist on type 'TaskLibAnswers'.

I haven't dug into why it cannot find stats yet.

**Comment by sebcaps** [comment]
I absolutely don't understand this error.... As my dev friend often says 'it works on my machine'.
I'll have a check if I can do sthg but I suspect a diff between my machine and build agent...

**Comment by sebcaps** [comment]
Definitely no idea about compilation issue. @carlo Can you have a look ?

**Comment by sebcaps** [comment]
Finally it's OK ;-)

**Comment by carlowahlstedt** [comment]
Not ignoring you here, just been really busy. I'll try to get back to looking at this over the weekend. I appreciate your work and patience on this.

File: .gitignore [hunk 1]
```diff
@@ -29,4 +29,8 @@ bld/
 */dist/
 node_modules
 .vscode
-*.vsix
\ No newline at end of file
+*.vsix
+*.map
+*.js
```
Comment (carlowahlstedt) [inline .gitignore]: I agree with removing these .js files, but we should probably remove them from source as well. The ignore won't ignore already committed files.

File: .gitignore [hunk 2]
```diff
@@ -29,4 +29,8 @@ bld/
 */dist/
 node_modules
 .vscode
-*.vsix
\ No newline at end of file
+*.vsix
+*.map
+*.js
+NewmanPostman/npm-debug.log
+.taskkey
```
Comment (carlowahlstedt) [inline .gitignore]: Similar to .js above. If we don't need the .taskkey file, then we should remove it from source as well.

File: NewmanPostman/npm-debug.log [hunk 3]
```diff
@@ -0,0 +1,48 @@
+0 info it worked if it ends with ok
```
Comment (carlowahlstedt) [inline NewmanPostman/npm-debug.log]: Do we need to remove this file from source as well?

**Comment by carlowahlstedt** [comment]
So...hadn't really used the PR changes before. I had started comments days ago, but didn't realize I had to submit them before you could see them. Either way, they are there now.

Overall everything looks really good and I think your tests are probably a big testament to that. I'm thinking we will release this to a new major version (so people have to manually upgrade) since there are several potential breaking changes here.

File: NewmanPostman/npm-debug.log (hunk 3 above)
Comment (sebcaps) [reply NewmanPostman/npm-debug.log]: Done, please consider PR#49

File: .gitignore (hunk 2 above)
Comment (sebcaps) [reply .gitignore]: Done, please consider PR#49

**Comment by sebcaps** [comment]
Ok with your comments.
I applied them in PR#49...

**Pull request closed without merging**